### ISSUE TITLE
Question about BCEWithLogitsLoss

### ISSUE THREAD

**Issue by ruohoruotsi** [chrisby/torchMTL#5]
Hi, thanks for your nice library. I like the way the tasks are laid out cleanly and easy to read.

So I have a `5`-task network that I'd like to build off of a [shared backbone](https://miro.medium.com/max/565/0*ZwCyanzdxYyWXLKK.png) ... pretty simple. The simplest, vanilla multi-task PyTorch (no backbone) code might look something like

```
model = nn.Linear(20, 5) # predict logits for 5 classes
x = torch.randn(1, 20)
y = torch.tensor([[1., 0., 1., 0., 0.]]) # get classA and classC as active

criterion = nn.BCEWithLogitsLoss()
optimizer = optim.SGD(model.parameters(), lr=1e-1)

for epoch in range(20):
    optimizer.zero_grad()
    output = model(x)
    loss = criterion(output, y)
    loss.backward()
    optimizer.step()
    print('Loss: {:.3f}'.format(loss.item()))
```

now using torchMTL ... I have my tasks setup like below with a number of tasks, [built on task-specific "stacks", sitting on a shared backbone](https://miro.medium.com/max/565/0*ZwCyanzdxYyWXLKK.png). Each task is a binary 0/1 decision, so a `Linear` layer with dimension=`1` for the output and `BCEWithLogitsLoss`. My question is the the correct setup, i.e. output-dimension==1?

```
...
    {
        'name': "Task2",
        'layers': Sequential(*[Linear(64, 1)]),
        'loss': BCEWithLogitsLoss(),
        'loss_weight': 1.0,
        'anchor_layer': 'StackB'
    },
    {
        'name': "Task3",
        'layers': Sequential(*[Linear(64, 1)]),
        'loss': BCEWithLogitsLoss(),
        'loss_weight': 1.0,
        'anchor_layer': 'StackC'
    },
```

The reason I ask the question, is that in my trivial example above. I use multi-hot encoded targets, `nn.BCE(WithLogits)Loss` and an output layer returning `[batch_size, nb_classes]` so `BCEWithLogitsLoss` processes all the output tasks at once, treating each output independently and so is suitable for multi-label classification use cases. So 1 criterion, 5 classes

In my torchMTL code, I'm not sure how to aggregate the outputs of each of the tasks in order to stack them together or even if I should, because it looks like I have 5 criterions and 5 classes ...which might work for regression, but my code is broken wrt to  `BCEWithLogitsLoss`. Do you have any guidance?  Thank you!

**Comment by ruohoruotsi** [comment]
I resolved the issue/question that I had. I can indeed use/track individual criterions separately. So by setting up my training example labels `y` to have an extra dimension via `unsqueeze(2)`, then I sliced `y` along the appropriate dimension when computing the per-task loss, i.e.

```
ith_loss = l_funcs[i](y_hat[i], y[:, i])
```

The choice to of how many "tasks per loss" to track/compute with `BCEWithLogitsLoss` (which can process multiple tasks at once) is still a design issue, but multiple `Linear` layer stacks with single output dimensions (as above are not limited by this framework. Closing the issue.
